### PR TITLE
Playlist

### PR DESCRIPTION
**Pull request by MuskanChhatrasal**
Done with playlist functionality.

### CODE REVIEW

File: src/Components/CardPlaylist/cardPlaylist.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import "./cardPlaylist.css";
+import { Link } from "react-router-dom";
+import { useHistory } from "../../Context/historyContext";
```
Comment (aayushjain-code) [inline src/Components/CardPlaylist/cardPlaylist.js:3]: You can use an index file in context file and import all Contexts at once

File: src/Components/CardPlaylist/cardPlaylist.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        </Link>
+        <div className="container-btn-playlist-card flex flex-column">
+          <div className="title-playlist">{playlist.title}</div>
+          <button
+            className="btn-remove-playlist btn-transparent"
+            onClick={() => removePlaylist(playlist?._id)}
+          >
+            <span
+              className="material-icons icon btn-transparent pdr-0-5"
+              style={{ color: "white", marginTop: "-0.2rem" }}
```
Comment (aayushjain-code) [inline src/Components/CardPlaylist/cardPlaylist.js:61]: Try to use class and avoid importing inline styling

File: src/Components/VideoCard/videoCard.js [hunk 3]
```diff
@@ -1,11 +1,40 @@
-import React, { useEffect, useState } from "react";
+import React, { useEffect, useState, useRef } from "react";
 import { Link, useLocation } from "react-router-dom";
 import { useHistory } from "../../Context/historyContext";
+import { useLikedVideo } from "../../Context/likeVideoContext";
```
Comment (aayushjain-code) [inline src/Components/VideoCard/videoCard.js:4]: You can use an index file in the context file and import all Contexts at once.
`import {useHistory ,useLikedVideo,usePlaylist  ,useWatchLater  } from "../../Contex/index `

File: src/Components/VideoCard/videoCard.js [hunk 4]
```diff
@@ -1,11 +1,40 @@
...
+  const {
+    addNewplaylistLoading,
+    addVideoToplaylistLoading,
+    removeVideoFromPlaylistLoading,
+    playlists,
+    playlistError,
+  } = playlistState;
+
   const location = useLocation();
   const trimHeading = (word, n) => {
```
Comment (aayushjain-code) [inline src/Components/VideoCard/videoCard.js:39]: You Can make a utility file and export it from there and import wherever needed

File: src/Components/VideoCard/videoCard.js [hunk 5]
```diff
@@ -107,6 +185,81 @@ const VideoCard = ({ video }) => {
...
+                <CardVideoPlaylist
+                  playlist={playlist}
+                  item={video}
+                  key={video._id}
+                />
+              );
+            })}
+        </div>
+
+        {/* <div className="playlist-second-div">
```
Comment (aayushjain-code) [inline src/Components/VideoCard/videoCard.js:216]: you Can remove commented code if not needed

File: src/Pages/Liked/liked.js [hunk 6]
```diff
@@ -1,11 +1,51 @@
...
+            <h1>No Videos added to Watch Later</h1>
+          )}
+
+          {LikedVideos.length === 0 && (
+            <h1
+              style={{
+                color: "white",
+                marginLeft: "15rem",
+                marginTop: "4rem",
+              }}
```
Comment (aayushjain-code) [inline src/Pages/Liked/liked.js:42]: Give a separate class for this to style

File: src/Pages/Playlist/playlist.css [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+.container-overlay-img-playlist-card {
+    width: 6rem;
+    height: 11rem;
+    position: absolute;
+    top: 0;
+    right: 0;
+    background-color: #85828280;
+}
+
+.btn-view-playlist {
```
Comment (aayushjain-code) [inline src/Pages/Playlist/playlist.css:45]: colors should be declared

File: src/Pages/Playlist/playlist.js [hunk 8]
```diff
@@ -1,12 +1,67 @@
...
+      <main className="main-playlist">
+        {playlists.length > 0 ? (
+          <h4
+            className="pdl-3 pdt-3 font-semibold"
+            style={{
+              marginTop: "-65rem",
+              marginLeft: "-3rem",
+              color: "white",
+              marginBottom: "-2rem",
+            }}
```
Comment (aayushjain-code) [inline src/Pages/Playlist/playlist.js:26]: Give separate Class to this

**Review by Developers-codz — APPROVED** [review]
UI looks amazing good work on code as well

**Review by Developers-codz — APPROVED** [review]
UI looks amazing good work on code as well

File: src/Pages/SingleVideoPlaylist/singleVideoPlaylist.js [hunk 9]
```diff
@@ -0,0 +1,119 @@
+import "../SingleVideo/singleVideo.css";
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { useSingleVideo } from "../../Context/singleVideoContext";
+import { usePlaylist } from "../../Context/playlistContext";
+import SingleVideoCard from "../../Components/SingleVideoCard/singleVideoCard";
+import Sidebar from "../../Components/Sidebar/sidebar";
+import { useWatchLater } from "../../Context/watchLaterContext";
+import VideoCard from "../../Components/VideoCard/videoCard";
```
Comment (aayushjain-code) [inline src/Pages/SingleVideoPlaylist/singleVideoPlaylist.js:9]: Give a default import for this

**Review by aayushjain-code — APPROVED** [review]
With few improvements you can merge it!

**Pull request merged**